--- a/luci-app-advanced/luasrc/model/cbi/advanced.lua
+++ b/luci-app-advanced/luasrc/model/cbi/advanced.lua
@@ -237,4 +237,47 @@ e.remove("/tmp/smartdns")
 end
 end
 end
+
+if nixio.fs.access("/etc/dnsmasq.conf")then
+s:tab("dnsmasqsconf",translate("DNSMASQ"),translate("本页是配置/etc/dnsmasq.conf, 包含Dnsmasq配置文档内容。应用保存后自动生效"))
+conf=s:taboption("dnsmasqsconf",Value,"dnsmasqsconf",nil,translate("开头的数字符号（＃）或分号的每一行（;）被视为注释；删除（;）启用指定选项。"))
+conf.template="cbi/tvalue"
+conf.rows=20
+conf.wrap="off"
+conf.cfgvalue=function(t,t)
+return e.readfile("/etc/dnsmasq.conf")or""
+end
+conf.write=function(a,a,t)
+if t then
+t=t:gsub("\r\n?","\n")
+e.writefile("/tmp/dnsmasq.conf",t)
+if(luci.sys.call("cmp -s /tmp/dnsmasq.conf /etc/dnsmasq.conf")==1)then
+e.writefile("/etc/dnsmasq.conf",t)
+luci.sys.call("/etc/init.d/dnsmasq restart >/dev/null")
+end
+e.remove("/tmp/dnsmasq.conf")
+end
+end
+end
+
+if nixio.fs.access("/etc/hosts")then
+s:tab("hostsconf",translate("HOSTS"),translate("本页是配置/etc/hosts, 包含Hosts配置文档内容。应用保存后自动生效"))
+conf=s:taboption("hostsconf",Value,"hostsconf",nil,translate("开头的数字符号（＃）或分号的每一行（;）被视为注释；删除（;）启用指定选项。"))
+conf.template="cbi/tvalue"
+conf.rows=20
+conf.wrap="off"
+conf.cfgvalue=function(t,t)
+return e.readfile("/etc/hosts")or""
+end
+conf.write=function(a,a,t)
+if t then
+t=t:gsub("\r\n?","\n")
+e.writefile("/tmp/hostsconf",t)
+if(luci.sys.call("cmp -s /tmp/hostsconf /etc/hosts")==1)then
+e.writefile("/etc/hosts",t)
+end
+e.remove("/tmp/hostsconf")
+end
+end
+end
 return m
