--- a/luci-app-rebootschedule/luasrc/controller/rebootschedule.lua
+++ b/luci-app-rebootschedule/luasrc/controller/rebootschedule.lua
@@ -4,8 +4,8 @@ function index()
 		return
 	end
 	
-	entry({"admin", "control"}, firstchild(), "Control", 44).dependent = false
-	entry({"admin", "control", "rebootschedule"}, cbi("rebootschedule"), "Timing setting", 20).dependent = true
+
+	entry({"admin", "system", "rebootschedule"}, cbi("rebootschedule"), _("定时任务"), 20)
 end
 
 
--- a/luci-app-rebootschedule/root/etc/init.d/rebootschedule
+++ b/luci-app-rebootschedule/root/etc/init.d/rebootschedule
@@ -20,7 +20,7 @@ if [ "`grep 'option enable .1.' /etc/config/rebootschedule 2>/dev/null`" ]; then
 			hour=`uci get rebootschedule.@crontab[$i].hour 2>/dev/null` || hour="*"; [ "$hour" = "0" ] && hour="00"
 			minute=`uci get rebootschedule.@crontab[$i].minute 2>/dev/null` || minute="00"; [ "$minute" = "0" ] && minute="00"
 			command=`uci get rebootschedule.@crontab[$i].command 2>/dev/null` || command="echo "Reboot schedule tested.""
-			echo "$minute $hour $day $month $week $command #rebootschedule" >> /etc/crontabs/root
+			echo -e "#rebootschedule\n$minute $hour $day $month $week $command" >> /etc/crontabs/root
 		fi
 	done
 	[ $mcronrst ] && /etc/init.d/cron restart

