--- a/luci-app-unblockneteasemusic/luasrc/controller/unblockneteasemusic.lua
+++ b/luci-app-unblockneteasemusic/luasrc/controller/unblockneteasemusic.lua
@@ -9,7 +9,7 @@ function index()
 	end
 
 	local page
-	page = entry({"admin", "services", "unblockneteasemusic"}, firstchild(), _("解除网易云音乐播放限制"), 50)
+	page = entry({"admin", "services", "unblockneteasemusic"}, firstchild(), _("网易云音乐"), 96)
 	page.dependent = false
 	page.acl_depends = { "luci-app-unblockneteasemusic" }
 
 --- a/luci-app-unblockneteasemusic/root/etc/init.d/unblockneteasemusic
+++ b/luci-app-unblockneteasemusic/root/etc/init.d/unblockneteasemusic
@@ -56,7 +56,6 @@ start_service()
 	config_get update_time "config" "update_time" "3"
 	sed -i "/$NAME/d" /etc/crontabs/root
 	is_enabled "config" "auto_update" && echo "0 ${update_time} * * * /usr/share/$NAME/update.sh update_core" >> "/etc/crontabs/root"
-	echo "*/5 * * * * /usr/share/$NAME/log_check.sh" >> "/etc/crontabs/root"
 	/etc/init.d/cron restart
 
 	[ ! -s "/usr/share/$NAME/core/app.js" ] && { rm -f "/usr/share/$NAME/local_ver"; sh "/usr/share/$NAME/update.sh" "update_core_non_restart"; }
