--- a/luci-app-rebootschedule/luasrc/controller/rebootschedule.lua
+++ b/luci-app-rebootschedule/luasrc/controller/rebootschedule.lua
@@ -4,8 +4,8 @@ function index()
 		return
 	end
 	
-	entry({"admin", "control"}, firstchild(), "Control", 44).dependent = false
-	entry({"admin", "control", "rebootschedule"}, cbi("rebootschedule"), "Timing setting", 20).dependent = true
+
+	entry({"admin", "system", "rebootschedule"}, cbi("rebootschedule"), _("定时任务"), 20).dependent = true
 end
 
 
